### PR TITLE
Issue 402

### PR DESCRIPTION
**Pull request by jeremiah-c-leary**
This is the implementation of issue #402 .

Am I doing something wrong?  I have an issue and a pull request.  This seems redundant.

### CODE REVIEW

File: docs/rule_severity.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+           "length_001":{
+               "severity": "Warning"
+           }
+       }
+   }
+
+Defining Severity Levels
+------------------------
+
+VSG supports defini
```
Comment (ericjmcd) [inline docs/rule_severity.rst:38]: Looks like you forgot to finish the comments here.

**Comment by ericjmcd** [comment]
I've tested this a little bit and it seems really good.  However, the junit output still counts all the findings are failures.  E.g. I set the default to a Guideline as warning and overrode 1 rule to be a Warning and the stdout matches:
```
architecture_026          | Guideline  |         48 | Align colon with right most colon.
variable_assignment_005   | Guideline  |        158 | Inconsistent alignment of ":=" in group of lines.
concurrent_005            | Warning    |        361 | Remove label on concurrent assignment.
```
However, the junit output has all failures.
```
<failure type="Failure">
      architecture_026: 48 : Align colon with right most colon.
...
```
I think extract_junit_testcase() needs to be updated?  I didn't want to look too deep in the code.  I tried to find a good spec on the JUnit XML but couldn't find anything concrete. This is the best I found:
[JUnit+Format](https://help.catchsoftware.com/display/ET/JUnit+Format)

There doesn't seem to be any support for warnings so I think for severities that aren't error/failures they should just be ignored in the JUnit output.

**Comment by jeremiah-c-leary** [comment]
> There doesn't seem to be any support for warnings so I think for severities that aren't error/failures they should just be ignored in the JUnit output.

What you would think about separating the warnings into the different severities in the JUnit XML file.  The thinking being you could have the CI tool ignore the severities you do not care about.  Although, that seems like extra work for the user.  So maybe just removing them from the JUnit XML file would be best.

File: docs/rule_severity.rst (hunk 1 above)
Comment (jeremiah-c-leary) [reply docs/rule_severity.rst:38]: agree, not sure what happened there.  Probably distracted by something shiny.

**Comment by jeremiah-c-leary** [comment]
@ericjmcd ,
I added the filtering of warnings from the JUnit file.  Could you update and check if it is working as you expect it work?

Thanks,

--Jeremy

**Comment by ericjmcd** [comment]
The test case classname is "failure" which is a bit odd (maybe change that just to "vsg"?) but there aren't any <failure> nodes now.  However, the cumulative failures at the <testsuite> level still shows the errors.

For what it's worth, I don't think I will end up using the unit output as the Jenkins plugins (xUnit) and the JUnit format don't support the severity construct so the errors alone aren't terribly valuable.  I think I'll write my own stdout parser so I can track/plot errors/warnings/guidelines/etc.  Of course, if time permits...

**Comment by jeremiah-c-leary** [comment]
Take a look at issue #417 .  I could catagorize by severity, or add the severity to each violation.

**Comment by jeremiah-c-leary** [comment]
> The test case classname is "failure" which is a bit odd (maybe change that just to "vsg"?) but there aren't any <failure> nodes now. However, the cumulative failures at the <testsuite> level still shows the failures (e.g. 26 files with Guideliens will show failures=26)..

I am a little unclear about what the issue is.  Could you elaborate or post some example JUnit files?

**Comment by jeremiah-c-leary** [comment]
Hey @ericjmcd ,

Curious if this is ready to merged or if there is anything else I need to do.

Thanks,

--Jeremy

**Pull request merged**